spec/core/link.fmf

```diff
@@ -0,0 +1,107 @@
+summary: Link related objects
+
+story:
+    As a user I want to link related tests, stories or external
+    issues so that I can easily find out what the test, plan or
+    story is addressing.
+
+description: |
+    The core attribute ``link`` is used to track relevant objects
+    such as user stories or external issues verified by the given
+    test, source code implementing particular story, blocking
+    issues or parent-child relations.
+
+    The value can be a single link or a list of links. There are
+    two options how to specify the ``target`` of the reference:
+
+    string
+        This can be either a ``name`` of the fmf node within the
+        same metadata tree, a file ``path`` from the tree root to
+        the application or documentation source code or a ``URL``
+        to an external issue tracker.
+
+    dict
+        In order to reference remote fmf objects use a dictionary
+        with the full `fmf identifier`__.
+
+    By default, the link between the two objects is interpreted in
+    a generic way as ``relates``. To explicitly define a different
+    connection to the target reference, use a dictionary and set
+    the key to one of the supported relations:
+
+    verifies
+        This object (a test or a plan) verifies functionality
+        defined in the target (a story, an issue, a bug).
+
+    verified-by
```

Comment:
I see bit confusion using `verifies` and counterpart `verified-by` and others.
I understand the meaning, and logic therem But:
 - this opposite side maybe adds there more complication in case it will be used nonexperienced used
 - it creates oriented graph instream of just one way of direction
 - it could be automatically transformed (some tooling to be able to convert `* -> *-by`)

Theoretically there is missed more complicated scenario  or documented something like, if possible:
```
/story:
  /story1:
    verifies: url://bugzilla
    verified-by: /test/test1
/test:
  /test1:
    test: runtest.sh
```

or this is not allowed?

I have fear that it will be very hard to implment the logic there in this case of vice-versa values.

Reply:
Regarding `verifies` and `verified-by`: We need both already now. Tangible examples are linking bugzilla from a test (should be soon supported by `tmt test import` as well) for the first one and linking test from a story (currently frequently used in tmt stories).

As far as implementation is concerned, I would start with implementing just one direction (the defined one) at the beginning and later extend with auto-creating the opposite links as well. This does not have to be completed in the first version, I believe.

Also `tmt test lint` (and its plan/story counterparts) could support checking (and possibly adding) the backlinks.

Reply:
Regarding the example above: I don't expect that story `verifies` a bug would be useful scenario. How do you propose to interpret this?

Reply:
Scenario (story) could be based on some bug (e.g. RFE or some jira task) so that scenario `verifies` the bug and when coverage for this scenario is created then scenario is `verified-by` some automated test. So from my perspective it makes sense. But I understand that it is not necessary, and could be described in another way, just this were example why I see difficult and complicated to implement this `*` `*-by` links.

Yes it make sense to implement just one way for now, just good to know if it is implemented already.

Reply:
I'd say, the fact that a story is based on a bug/issue does not mean the story verifies it. In this particular case I would suggest to use `relates` for the connection between story and bug and `verifies` from the test. It could look like this:

```yaml
/some/story:
    link: https://bugzilla.redhat.com/show_bug.cgi?id=1234

/some/test:
    link:
      - verifies: /some/story
      - verifies: https://bugzilla.redhat.com/show_bug.cgi?id=1234
```

I think we have enough flexibility with the currently proposed relations (at least to get started).